app/grandchallenge/evaluation/migrations/0077_create_evaluation_input_set.py

```diff
@@ -0,0 +1,117 @@
+# Generated by Django 4.2.20 on 2025-03-19 08:49
+import uuid
+
+import django.db.models.deletion
+from django.db import migrations, models
+
+from grandchallenge.evaluation.models import (
+    get_existing_evaluation_input_set_for_inputs,
+)
+
+
+def update_evaluation_inputs(apps, _schema_editor):
+    Evaluation = apps.get_model("evaluation", "Evaluation")  # noqa: N806
+    EvaluationInputSet = apps.get_model(  # noqa: N806
+        "evaluation", "EvaluationInputSet"
+    )
+
+    batch_size = 5000
+
+    evaluations = Evaluation.objects.prefetch_related("inputs").iterator(
+        chunk_size=batch_size
+    )
+    updated_evaluations = []
+
+    for evaluation in evaluations:
+        inputs = evaluation.inputs.all()
+
+        input_set = get_existing_evaluation_input_set_for_inputs(
+            model=EvaluationInputSet,
+            inputs=inputs,
+        )
+        if not input_set:
+            input_set = EvaluationInputSet.objects.create()
+            input_set.values.set(inputs)
+
+        evaluation.input_set = input_set
+        updated_evaluations.append(evaluation)
+
+        if len(updated_evaluations) >= batch_size:
+            Evaluation.objects.bulk_update(updated_evaluations, ["input_set"])
+            updated_evaluations = []
+
+    if updated_evaluations:
+        Evaluation.objects.bulk_update(evaluations, ["input_set"])
```

Comment:
This then updates all evaluations? Wasn't the point of the iterator not to have them in memory?